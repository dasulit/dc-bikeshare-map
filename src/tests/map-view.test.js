import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import MapView from '../components/map-view';

describe('MapView', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      React.createElement(MapView, {
        bikeData: { features: [] }
      })
    );
  });

  test('renders', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('only adds source when bike data is available', () => {
    wrapper.instance().addBikeDataToMap = jest.fn();
    expect(wrapper.instance().addBikeDataToMap).toHaveBeenCalledTimes(0);
    wrapper.setProps({
      bikeData: { features: [('id': 'feature-1')] }
    });
    wrapper.update();
    expect(wrapper.instance().addBikeDataToMap).toHaveBeenCalledTimes(1);
  });
});
