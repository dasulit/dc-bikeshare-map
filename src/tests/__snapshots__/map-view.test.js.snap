// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapView renders 1`] = `
<div
  id="map"
/>
`;
